mesonbuild/compilers/compilers.py

```diff
@@ -505,42 +511,54 @@ def get_display_language(cls) -> str:
     def get_default_suffix(self) -> str:
         return self.default_suffix
 
-    def get_define(self, dname, prefix, env, extra_args, dependencies) -> T.Tuple[str, bool]:
+    def get_define(self, dname: str, prefix: str, env: 'Environment',
+                   extra_args: T.Sequence[str], dependencies: T.Sequence['Dependency'],
+                   disable_cache: bool = False) -> T.Tuple[str, bool]:
```

Comment:
Should we convert all these to `@abc.abstractmethod`?

Reply:
The problem with abstractmethod is children cannot be instantiated without a concrete implementation, so we'd have to reimplement the same "not supported" method for every language. I think e should just use abstractmethod for cases where every language needs to implement the method